pkg/util/goroutinemap/goroutinemap.go

```diff
@@ -103,7 +107,8 @@ func (grm *goRoutineMap) Run(
 
 	grm.operations[operationName] = operation{
 		operationPending: true,
-		expBackoff:       existingOp.expBackoff,
+		expBackoff: exponentialbackoff.
+			ExponentialBackoff{MaxDurationBeforeRetry: grm.volumeOperationMaxBackoff},
```

Comment:
Why is this changing from reusing the `existingOp` to constructing a new object?

Reply:
Since we have changed the struct `ExponentialBackoff` and added `MaxDurationBeforeRetry`, when an operation is not existed, we should construct a new one using a specified value. If not, the default value(0) would be applied.

Reply:
this still doesn't look right. this line can be executed even when existingOp already existed. if you want to initialize in the `!exists` case, that need to be done in an `else` block on line 106